pkgs/tools/networking/maxscale/default.nix

```diff
@@ -0,0 +1,133 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, gcc, glibc
+, bison2, curl, flex, gperftools, jansson, jemalloc, kerberos, lua, mariadb
+, ncurses, openssl, pcre, pcre2, perl, rabbitmq-c, sqlite, tcl
+, libaio, libedit, libtool, libui, libuuid, zlib
+}:
+
+let
+  libPath1 = stdenv.lib.makeLibraryPath [ curl gcc.cc.lib glibc jemalloc kerberos openssl pcre2 zlib ];
+  libPath2 = stdenv.lib.makeLibraryPath [ glibc libedit ncurses ];
+  libPath3 = stdenv.lib.makeLibraryPath [ curl gcc.cc.lib glibc jemalloc kerberos libuuid openssl pcre2 zlib ];
+  libPath4 = stdenv.lib.makeLibraryPath [ curl gcc.cc.lib glibc jansson jemalloc kerberos libedit libuuid lua openssl pcre2 rabbitmq-c sqlite zlib ];
+in
+
+stdenv.mkDerivation rec {
+  name = "maxscale-${version}";
+  version = "2.1.13";
+
+  src = fetchFromGitHub {
+    owner = "mariadb-corporation";
+    repo = "MaxScale";
+    rev = "${name}";
+    sha256 = "0bwm98mbjay24vf95rk4slvjqxqblx14hvv75dpp1lhfhw25hhrf";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+
+  buildInputs = [
+    bison2 curl flex gperftools jansson jemalloc kerberos lua mariadb.connector-c
+    ncurses openssl pcre pcre2 perl rabbitmq-c sqlite tcl
+    libaio libedit libtool libui libuuid zlib
+  ];
+
+  patches = [ ./getopt.patch ];
+
+  preConfigure = ''
+    for i in `grep -l -R '#include <getopt.h>' .`; do
+      substituteInPlace $i --replace "#include <getopt.h>" "#include <${glibc.dev}/include/getopt.h>"
+    done
+ '';
+
+  cmakeFlags = [
+    "-DUSE_C99=YES"
+    "-DDEFAULT_ADMIN_USER=root"
+    "-DWITH_MAXSCALE_CNF=YES"
+    "-DSTATIC_EMBEDDED=YES"
+    "-DBUILD_RABBITMQ=YES"
+    "-DBUILD_BINLOG=YES"
+    "-DBUILD_CDC=NO"
+    "-DBUILD_MMMON=YES"
+    "-DBUILD_LUAFILTER=YES"
+    "-DLUA_LIBRARIES=${lua}/lib"
+    "-DLUA_INCLUDE_DIR=${lua}/include"
+    "-DGCOV=NO"
+    "-DWITH_SCRIPTS=OFF"
+    "-DBUILD_TESTS=NO"
+    "-DBUILD_TOOLS=NO"
+    "-DPROFILE=NO"
+    "-DWITH_TCMALLOC=YES"
+    "-DWITH_JEMALLOC=YES"
+    "-DINSTALL_EXPERIMENTAL=YES"
+    "-DTARGET_COMPONENT=all"
+  ];
+
+  CFLAGS = "-std=gnu99";
+
+  enableParallelBuilding = true;
+
+  postInstall = ''
+    patchelf --set-rpath ${libPath1}:${mariadb.connector-c}/lib/mariadb:$out/lib/maxscale $out/bin/dbfwchk
+    patchelf --set-rpath ${libPath2}:$out/lib/maxscale $out/bin/maxadmin
```

Comment:
I wonder, is the primary goal here to *add* `$out/lib/...` to the rpath? If so I think you could profitably change these calls to include `patchelf --print-rpath ...` instead of specifying the libpath yourself.

Reply:
This method does not work for binary files:

Reply:
?

Reply:
Well, rpath wouldn't work at all then. See https://github.com/NixOS/nixpkgs/blob/8de132f52da85ee273823c1c89c1be8b5670b906/pkgs/applications/backup/deja-dup/default.nix#L53 for an example of what I'm talking about.

Reply:
changed to - 
`patchelf --set-rpath "$(patchelf --print-rpath "$out/bin/maxscale"):$out/lib/maxscale" "$out/bin/maxscale"`
Not worked - 
`ldd /nix/store/hn0lnffvfzqy0bnlmh30kgvxr7w8sxrq-maxscale-2.1.13/bin/maxscale`
`        not a dynamic executable`